File: backends/php-laravel/app/Domain/Invitation/Repositories/InvitationAuthRepository.php
<?php

/**
 * This is a program developed by BobTabo.
 *
 * Copyright (c) 2026 BobTabo. All Rights Reserved.
 */

declare(strict_types=1);

namespace App\Domain\Invitation\Repositories;

/**
 * 招待認証Repositoryのインターフェースです。
 *
 * @author Satoshi Nagashiba <satoshi.nagashiba@gmail.com>
 * @package App\Domain\Invitation\Repositories
 */
interface InvitationAuthRepository
{
    /**
     * 招待トークンとロールを一時保存します。
     *
     * @param string $token 招待トークン
     * @param int $role 権限（1=管理者, 2=メンバー）
     * @param int $ttl 有効期限（秒）
     */
    public function store(string $token, int $role, int $ttl): void;

    /**
     * 招待トークンに紐づくロールを取得します。
     *
     * @param string $token 招待トークン
     * @return int|null ロール、未保存の場合 null
     */
    public function find(string $token): ?int;

    /**
     * 招待トークンを削除します。
     *
     * @param string $token 招待トークン
     */
    public function remove(string $token): void;
}

File: backends/java-spring/src/main/java/com/authorization/domain/invitation/repositories/InvitationAuthRepository.java
/**
 * This is a program developed by BobTabo.
 *
 * Copyright (c) 2026 BobTabo. All Rights Reserved.
 */
package com.authorization.domain.invitation.repositories;

/**
 * 招待認証Repositoryのインターフェースです。
 *
 * @author Satoshi Nagashiba <satoshi.nagashiba@gmail.com>
 */
public interface InvitationAuthRepository {

    /**
     * 招待トークンとロールを一時保存します。
     *
     * @param token 招待トークン
     * @param role 権限（1=管理者, 2=メンバー）
     * @param ttl 有効期限（秒）
     */
    void store(String token, int role, int ttl);

    /**
     * 招待トークンに紐づくロールを取得します。
     *
     * @param token 招待トークン
     * @return ロール、未保存の場合 null
     */
    Integer find(String token);

    /**
     * 招待トークンを削除します。
     *
     * @param token 招待トークン
     */
    void remove(String token);
}
